go.mod

```diff
@@ -1,4 +1,4 @@
-module github.com/spiceai/gospice/v6
+module github.com/spiceai/gospice/v7
 
 go 1.22
```

Comment:
If we are upgrading major version, we should also upgrade Go (1.23) and Arrow.

Reply:
I'll be updating dependencies in another PR before the next SDK release

Reply:
I think Go modules are published as soon as the repo is updated, so it would need to be in this PR

Reply:
We should update the docs to have `go get` retrieve from a release tag rather than directly from `trunk`.

https://github.com/spiceai/gospice/issues/59

In the meantime, I'll update the dependencies in this PR.